### PR TITLE
add types for MachineDeployment provider configuration

### PR DESCRIPTION
**Pull request by xh3b4sd**
The types for the Cluster provider config are already there. Here are the types for the MachineDeployment provider config.

### CODE REVIEW

File: pkg/apis/cluster/v1alpha1/aws_machine_deployment_status_types.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	Provider          AWSMachineDeploymentStatusProvider `json:"provider" yaml:"provider"`
+}
+
+type AWSMachineDeploymentStatusCluster struct {
+	ID string `json:"id" yaml:"id"`
+}
+
+type AWSMachineDeploymentStatusNodePool struct {
+	ID       string                                      `json:"id" yaml:"id"`
+	Scaling  AWSMachineDeploymentStatusNodePoolScaling   `json:"nodePool" yaml:"nodePool"`
```
Comment (oponder) [inline pkg/apis/cluster/v1alpha1/aws_machine_deployment_status_types.go]: Should this be `json:"scaling" yaml:"scaling:` ? based on the pattern I see so far, and the example above.

File: pkg/apis/cluster/v1alpha1/aws_machine_deployment_status_types.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	Subnet AWSMachineDeploymentStatusProviderAvailabilityZoneSubnet `json:"subnet" yaml:"subnet"`
+}
+
+type AWSMachineDeploymentStatusProviderAvailabilityZoneSubnet struct {
+	Private AWSMachineDeploymentStatusProviderAvailabilityZoneSubnetPrivate `json:"private" yaml:"private"`
+	Public  AWSMachineDeploymentStatusProviderAvailabilityZoneSubnetPublic  `json:"public" yaml:"public"`
+}
+
+type AWSMachineDeploymentStatusProviderAvailabilityZoneSubnetPrivate struct {
+	CIDR string `json:"string" yaml:"string"`
```
Comment (oponder) [inline pkg/apis/cluster/v1alpha1/aws_machine_deployment_status_types.go]:
```suggestion
	CIDR string `json:"cidr" yaml:"cidr"`
```

File: pkg/apis/cluster/v1alpha1/aws_machine_deployment_status_types.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+type AWSMachineDeploymentStatusProviderAvailabilityZoneSubnetPublic struct {
+	CIDR string `json:"string" yaml:"string"`
```
Comment (oponder) [inline pkg/apis/cluster/v1alpha1/aws_machine_deployment_status_types.go]:
```suggestion
	CIDR string `json:"cidr" yaml:"cidr"`
```

File: pkg/apis/cluster/v1alpha1/aws_machine_deployment_status_types.go (hunk 1 above)
Comment (oponder) [inline pkg/apis/cluster/v1alpha1/aws_machine_deployment_status_types.go]:
```suggestion
	Scaling  AWSMachineDeploymentStatusNodePoolScaling   `json:"scaling" yaml:"scaling"`
```

**Review by oponder — CHANGES_REQUESTED** [review]
I'm not deep into the topic, so I just looked for typos or anything I could find, noticed some inconsistencies, forgive me if they are intentional.

File: pkg/apis/cluster/v1alpha1/aws_machine_deployment_spec_types.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+// extension.
+//
+//     kind: AWSMachineDeploymentSpec
+//     apiVersion: cluster.giantswarm.io/v1alpha1
+//     metadata:
+//       name: al9qy
+//     nodePool:
+//       description: my fancy node pool
+//       machine:
+//         dockerVolumeSizeGB: 100
```
Comment (kopiczko) [inline pkg/apis/cluster/v1alpha1/aws_machine_deployment_spec_types.go]: Isn't it an idea to allow specify unit in the value like with limits and requests, i.e. 100G?

**Review by kopiczko — APPROVED** [review]

File: pkg/apis/cluster/v1alpha1/aws_machine_deployment_spec_types.go (hunk 4 above)
Comment (xh3b4sd) [reply pkg/apis/cluster/v1alpha1/aws_machine_deployment_spec_types.go]: I think disk sizes are most useful and most simple at the same time to be configured in GB. I think that was a conscious decision from SIG UX back in the day.

File: pkg/apis/cluster/v1alpha1/aws_machine_deployment_status_types.go (hunk 1 above)
Comment (xh3b4sd) [reply pkg/apis/cluster/v1alpha1/aws_machine_deployment_status_types.go]: Thank you eagle eye. I missed the tags. Applied your suggestions.

**Review by oponder — APPROVED** [review]

**Pull request merged**